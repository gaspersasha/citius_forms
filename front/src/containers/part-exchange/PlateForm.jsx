import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { InputText, Button } from '~components';
import { INPUT_STATUS } from '~constants';
import {
  checkNotEmpty,
  checkUKPlate,
  endpointFetch,
  makeQueryString,
  GTM_EVENT,
} from '~utils';
import s from './styles/part-exchange.module.sass';

const PlateForm = ({
  setData,
  payData,
  setNextStep,
  urlParams,
  leadgen,
  scroll,
  deleteEmptyUrlParams,
  setIsLoading,
}) => {
  const [state, setState] = useState({
    value: payData.registration,
    status: INPUT_STATUS.DEFAULT,
  });

  const router = useRouter();

  useEffect(() => {
    setData({ registration: state.value });
  }, [state.value]);

  const submitRegistration = () => {
    setNextStep(0);
    setIsLoading(true);

    endpointFetch('checkPlatePartEx', {
      ...deleteEmptyUrlParams(urlParams),
      ...payData,
    })
      .then((data) => {
        setData({ ...data, step: 1 });
        scroll();
        setIsLoading(false);

        GTM_EVENT('ssAddPartExchange');
      })
      .catch(console.log);
  };

  const handleChange = ({ target: { value } }) => {
    if (value.length < 10) setState({ ...state, value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const { value } = state;

    if (!value || !checkNotEmpty(value) || !checkUKPlate(value)) {
      return setState({ ...state, status: INPUT_STATUS.INVALID });
    }

    setState({ ...state, status: INPUT_STATUS.DEFAULT });
    submitRegistration();
  };

  const handleSkip = (e) => {
    e.preventDefault();

    GTM_EVENT('ssPartExchangeSkipped');

    const url = makeQueryString('terms', {
      quoteId: router.query.quoteId || '',
      quoteItemId: router.query.quoteItemId || '',
      productAdvertId: router.query.productAdvertId || '',
      leadgen,
    });

    router.push(url);
  };

  return (
    <form data-id="plate-form" onSubmit={handleSubmit}>
      <h5 className={s.heading}>Add part exchange?</h5>
      <h5 className={s.description}>
        Enter your reg below to get your estimated part exchange price
      </h5>
      {state.status === INPUT_STATUS.INVALID && (
        <div className={s.errorText}>*Please enter valid UK plate number</div>
      )}
      <div className={s.plate}>
        <InputText
          value={state.value}
          status={state.status}
          onChange={handleChange}
          label=""
          name="plate"
          placeholder="e.g. AB12CDE"
        />
      </div>
      <p className={s.infoBox}>
        By using this service, you agree that your vehicle registration details
        will be passed to a third party, in order to look up the vehicle
        details.
      </p>
      <div className={s.alignCenter}>
        <Button type="submit" name="partex__submit-btn">
          Next
        </Button>
        <span
          className={s.skipPX}
          onClick={handleSkip}
          role="button"
          tabIndex={0}
        >
          Skip part exchange
        </span>
      </div>
    </form>
  );
};

PlateForm.propTypes = {
  urlParams: PropTypes.object.isRequired,
  setData: PropTypes.func.isRequired,
  payData: PropTypes.any.isRequired,
  setNextStep: PropTypes.func.isRequired,
  deleteEmptyUrlParams: PropTypes.func.isRequired,
  setIsLoading: PropTypes.func.isRequired,
  scroll: PropTypes.func,
  leadgen: PropTypes.string,
};

PlateForm.defaultProps = {
  scroll: PropTypes.func,
  leadgen: '0',
};

export default PlateForm;
