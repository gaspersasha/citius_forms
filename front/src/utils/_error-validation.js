// /*
//  * accepts data with string type and returns boolean
//  * The following email validation requirements have been taken from
//  * https://help.salesforce.com/articleView?id=000321158&type=1&mode=1
//  */
// export const isValidEmail = email => {
//   if (!email || !email.length) return false;
//   if (email.indexOf('@') < 0) return false;

//   const [, domain] = email.split('@');

//   if (!domain || !domain.length) return false;

//   // manual validation for '..', cause it fails to work in Regexp in Safari and FF
//   if (domain.indexOf('..') > -1) {
//     return false;
//   }

//   // REGEXP TRANSCRIPTION
//   // LOCAL PART:
//   // ^(?!\.) => not start with dot
//   // [\w\/\-+.!#$%&'*=?^`{|}~]+ => group of chars from list, any amount

//   // DOMAIN PART:
//   // @([\w-][.]?)+ => group of chars with hyphen and  zero or one dot, any amount
//   // \.(?<!\.\.) => dot, but not two dots in a row
//   // [A-Za-z0-9]{2,20} => group of chars (min 2, max 20) for root domain

//   return /^(?!\.)[\w/\-+.!#$%&'*=?^`{|}~]+@([\w-][.]?)+\.[A-Za-z0-9]{2,20}$/.test(email);
// };

// export default function errorValidation(values) {
//   const errors = {};

//   // validate email
//   if (!isValidEmail(values.email)) {
//     errors.email = 'Invalid email address';
//   }

//   // validate password
//   if (!isValidEmail(values.password)) {
//     errors.password = 'Invalid password';
//   }

//   return errors;
// }

// /*
//  * Password validation for Profile form (Can be empty)
//  * Contain at least 8 characters
//  * contain at least 1 number
//  * contain at least 1 lowercase character (a-z)
//  * contain at least 1 uppercase character (A-Z)
//  * contains only 0-9a-zA-Z
//  */
// export const c = value => {
//   // if (typeof value === 'string') value = value.trim();
//   const regexp = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*[!@#%&*?Â£]).{8,}$/;

//   return regexp.test(value.trim());
// };
