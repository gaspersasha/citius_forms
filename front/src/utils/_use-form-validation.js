// import { useState, useEffect } from 'react';

// export default function useFormValidation(initialState, validate) {
//   const [values, setValues] = useState(initialState);
//   const [errors, setErrors] = useState({});
//   const [isSubmitting, setSubmitting] = useState(false);

//   useEffect(() => {
//     if (!isSubmitting) return;

//     const noErrors = Object.keys(errors).length === 0;

//     if (noErrors) {
//       console.log('submitting form');
//     } else {
//       console.log('NOT submitting form');
//     }

//     setSubmitting(false);
//   }, [errors]);

//   function handleChange(event) {
//     setValues({
//       ...values,
//       [event.target.name]: event.target.value,
//     });
//   }

//   function handleSubmit(event) {
//     event.preventDefault();
//     event.stopPropagation();
//     setSubmitting(true);
//     setErrors(validate(values));
//   }

//   function handleBlur() {
//     setErrors(validate(values));
//   }

//   return {
//     handleChange,
//     handleBlur,
//     handleSubmit,
//     values,
//     errors,
//     isSubmitting,
//   };
// }
