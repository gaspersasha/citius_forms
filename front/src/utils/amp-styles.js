/* TODO: suggestion for refactoring in ISSUES on github this project */
/* eslint max-len: ["error", { "ignoreTemplateLiterals": true }] */
export const ampStyles = `
  html, body, div, span,
  h1, h2, h3, h4, h5, h6, p,
  a, img, ul, li, label,
  footer, header, menu, nav {
    margin: 0;
    padding: 0;
    border: 0;
    font: inherit;
    font-size: 100%;
    vertical-align: baseline;
  }

  html {
    line-height: 1;
  }

  html {
    line-height: 1.625em;
    font-size: 100%;
    color: #333333;
    font-family: "sofia-pro", sans-serif;
  }

  ol, ul {
    list-style: none;
  }

  header, main,
  menu, section {
    display: block;
  }

  @font-face {
    font-family: 'sofia-pro';
    src:
      url("https://use.typekit.net/af/02ad94/00000000000000003b9b46f3/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3") format("woff2"),
      url("https://use.typekit.net/af/02ad94/00000000000000003b9b46f3/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3") format("woff"),
      url("https://use.typekit.net/af/02ad94/00000000000000003b9b46f3/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3") format("opentype");
    font-display: auto;
    font-style: normal;
    font-weight: 700;
  }
  @font-face {
    font-family: 'sofia-pro';
    src:
      url("https://use.typekit.net/af/ed85d3/00000000000000003b9b46f4/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i7&v=3") format("woff2"),
      url("https://use.typekit.net/af/ed85d3/00000000000000003b9b46f4/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i7&v=3") format("woff"),
      url("https://use.typekit.net/af/ed85d3/00000000000000003b9b46f4/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i7&v=3") format("opentype");
    font-display: auto;
    font-style: italic;
    font-weight: 700;
  }
  @font-face {
    font-family: 'sofia-pro';
    src:
      url("https://use.typekit.net/af/4cc789/00000000000000003b9b46ed/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n4&v=3") format("woff2"),
      url("https://use.typekit.net/af/4cc789/00000000000000003b9b46ed/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n4&v=3") format("woff"),
      url("https://use.typekit.net/af/4cc789/00000000000000003b9b46ed/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n4&v=3") format("opentype");
    font-display: auto;
    font-style: normal;
    font-weight: 400;
  }
  @font-face {
    font-family: 'sofia-pro';
    src:
      url("https://use.typekit.net/af/964ed8/00000000000000003b9b46ee/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i4&v=3") format("woff2"),
      url("https://use.typekit.net/af/964ed8/00000000000000003b9b46ee/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i4&v=3") format("woff"),
      url("https://use.typekit.net/af/964ed8/00000000000000003b9b46ee/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=i4&v=3") format("opentype");
    font-display: auto;
    font-style: italic;
    font-weight: 400;
  }

  .page-header {
    height: 60px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding-left: 15px;
    background: #07dda4;
    background: linear-gradient(to right, #03d397 50%, #07dda4 50% 100%);
  }

  .amp-sidebar-mask {
      background: none;
  }

  .menu-toggle {
    position: absolute;
    left: auto;
    right: 0;
    width: 14px;
    height: 12px;
    display: flex;
    flex-direction: column;
    box-sizing: content-box;
    padding: 24px 80px 24px 20px;
    justify-content: space-between;
    transform: rotate(0deg);
    transition: .5s ease-in-out;
    cursor: pointer;
    z-index: 1;
  }

  .menu-toggle:after {
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    width: 80px;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    color: #0f0f26;
    content: "Menu";
  }

  .menu-toggle span:nth-child(1) {
    top: 0px;
    transform-origin: left center;
  }
  .menu-toggle span:nth-child(2) {
    top: 50%;
    transform-origin: left center;
  }
  .menu-toggle span:nth-child(3) {
    top: 100%;
    transform-origin: left center;
  }

  amp-sidebar[open] ~ header .menu-toggle span:nth-child(1) {
    transform: rotate(45deg);
    margin-left: 1px;
  }

  amp-sidebar[open] ~ header .menu-toggle span:nth-child(2) {
    width: 0%;
    margin-left: 50%;
    opacity: 0;
  }

  amp-sidebar[open] ~ header .menu-toggle span:nth-child(3) {
    transform: rotate(-45deg);
    margin-left: 1px;
  }

  .menu-toggle span {
    display: block;
    height: 2px;
    width: 100%;
    border-radius: 9px;
    opacity: 1;
    background-color: #0f0f26;
    margin-left: 0;
    transform: rotate(0deg);
    transition: .25s ease-in-out;
  }

  amp-sidebar {
    top: 60px;
    max-width: 100vw;
    height: unset;
    background-color: #ffffff;
    padding-top: 70px;
    text-align: left;
    box-shadow: 0 0 10px rgb(0 0 0 / 50%);
  }

  header {
    margin-bottom: 0.8125em;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 60px;
  }

  header.menu {
    width: 100vw;
    height: 40px;
    background-color: #ffffff;
    padding-right: 0;
    border: 0;
    box-sizing: border-box;
  }

  header.menu:focus {
    outline:0;
  }

  amp-sidebar section:nth-child(6) {
    position: absolute;
    top: -70px;
    height: 55px;
    text-align: center;
  }

  amp-sidebar section:nth-child(6) header {
    height: 55px;
    width: 90vw;
    left: 5vw;
    border-bottom: 1px solid #0f0f26;
  }

  amp-sidebar section:nth-child(6) header a:before {
    content: '';
    width: 19px;
    height: 100%;
    background-repeat: no-repeat;
    background-position: center;
    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 513.64 513.64'%3E%3Cpath d='M499.66 376.96l-71.68-71.68c-25.6-25.6-69.12-15.359-79.36 17.92-7.68 23.041-33.28 35.841-56.32 30.72-51.2-12.8-120.32-79.36-133.12-133.12-7.68-23.041 7.68-48.641 30.72-56.32 33.28-10.24 43.52-53.76 17.92-79.36l-71.68-71.68c-20.48-17.92-51.2-17.92-69.12 0L18.38 62.08c-48.64 51.2 5.12 186.88 125.44 307.2s256 176.641 307.2 125.44l48.64-48.64c17.921-20.48 17.921-51.2 0-69.12z' fill='%231c1836' data-original='%23000000' xmlns='http://www.w3.org/2000/svg'/%3E%3C/svg%3E");
    background-size: 19px;
    margin-right: 10px;
  }

  amp-sidebar section ul.menu {
    min-width: 60vw;
    list-style: none;
    margin: 0;
  }

  amp-sidebar section ul.menu li.child-element {
    text-align: center;
    background-color: #ffffff;
  }

  amp-sidebar section ul.menu li.child-element a {
    color: #0f0f26;
    border: none;
    text-align: left;
    font-size: 16px;
    padding-left: 30px;
  }

  amp-sidebar section header.i-amp-html-accordion-header a,
  amp-sidebar section header.i-amp-html-accordion-header span {
    display: block;
    height: 40px;
    padding: 5px 15px;
    font-size: 16px;
    color: #0f0f26;
    text-decoration: none;
  }

  amp-sidebar section header.i-amp-html-accordion-header span:focus {
    outline: 0;
  }

  amp-sidebar section ul.menu li a, amp-sidebar section ul.menu li span {
    display: block;
    padding: 5px;
    text-decoration: none;
  }

  amp-sidebar section:nth-child(6) header a {
    height: 55px;
    line-height: 45px;
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
  }

  amp-sidebar section header.i-amp-html-accordion-header span:after {
    content: '';
    opacity: 1;
    transition: 0.2s transform, 0.6s opacity;
    border: 3px solid #0f0f26;
    border-bottom: 0;
    border-left: 0;
    width: 8px;
    height: 8px;
    position: absolute;
    right: 20px;
    top: 0;
    bottom: 0;
    margin: auto;
    transform: rotate(135deg);
    box-sizing: border-box;
  }

  amp-sidebar section[expanded] header.menu span:after,
  amp-sidebar section[expanded] ul.menu span:after {
    transform: rotate(315deg);
  }
  amp-sidebar[open] ~ header .menu-toggle:after {
    content: 'Close';
  }

  .separator {
    height: 54px;
    width: 100%;
    backgroundColor: transparent;
  }
`;
