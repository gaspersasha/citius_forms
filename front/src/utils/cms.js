/* eslint-disable no-use-before-define */
// const { NEXT_PUBLIC_DEVELOPMENT: isDev } = process.env;
import { ENV } from '~constants';

const { IS_PROD_ENV, IS_LOCAL_ENV } = ENV;

// args: relative url
export function getInfo(rurl) {
  let ENV_URL = 'https://buyacar.test2.didev.co.uk';

  if (IS_PROD_ENV) ENV_URL = 'https://buyacar.co.uk';
  if (IS_LOCAL_ENV) ENV_URL = '/_cms';
  const middle = IS_LOCAL_ENV ? '' : '/_content_api/v1?url=';
  const url = `${ENV_URL}${middle}${rurl}`;

  return fetch(url)
    .then((res) => res.json())
    .then(parse)
    .catch((err) => {
      console.log(
        `SOMETHING WRONG WITH CMS CONNECTION AT: ${ENV_URL} - ${err}`
      );

      return {
        title: 'BuyaCar: the easy way to get a great car deal',
        page_type: '',
        meta: {
          page_title: 'BuyaCar: the easy way to get a great car deal',
          description: 'Buyacar online with no hussle',
          rights: 'Autovia Group Limited, licensed by Felden',
          twitter: {
            title: 'BuyaCar: the easy way to get a great car deal',
            card: 'photo',
            site: '@buy_a_car',
            url: 'https://website',
          },
        },
      };
    });
}

// input object - Drupal Response
function parse({ Title, PageType, Meta }) {
  return {
    title: Title,
    page_type: PageType,
    meta: {
      page_title: Meta.PageTitle,
      description: Meta.Description,
      rights: Meta.Rights,
      canonical: Meta.Canonical,
      twitter: {
        title: Meta.Twitter.Title,
        card: Meta.Twitter.Card,
        site: Meta.Twitter.Site,
        url: Meta.Twitter.Url,
      },
    },
  };
}
