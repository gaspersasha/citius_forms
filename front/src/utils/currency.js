// TODO: try remove currency formatter dep in fav of Intl
import currencyFormatter from 'currency-formatter';

const currencyCode = () => 'GBP';

// https://github.com/smirzaei/currency-formatter/issues/31
const precision = (value) => (value % 1 === 0 ? 0 : null);

export const currencyFormat = (value = 0) =>
  currencyFormatter.format(value, {
    code: currencyCode(),
    precision: precision(value),
  });
// end TODO

export const formatNumber = (number) =>
  new Intl.NumberFormat().format(Number(number));
export const IntlGBP = new Intl.NumberFormat('en-EN', {
  style: 'currency',
  currency: 'GBP',
  minimumFractionDigits: 0,
  maximumFractionDigits: 0,
});
