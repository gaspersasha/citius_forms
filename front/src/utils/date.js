// eslint-disable-next-line import/no-extraneous-dependencies
import { parse, format } from 'date-fns';

// Format from java will always be Wed Jun 13 10:56:54 BST 2018
const dateFormat = 'ddd MMM DD HH:mm:ss zzz YYYY';

export const removeBST = (s) => s && s.replace('BST ', '');

export const formatDate = (fullDate = '') =>
  format(
    parse(removeBST(fullDate.toString()), dateFormat, new Date()),
    'DD.MM.YY'
  );

export const formatTime = (fullDate = '') =>
  format(parse(removeBST(fullDate.toString()), dateFormat), 'h:mma');

/**
 * returns {string} readable date in format "6th September 2021"
 * @param {string} inputDate - any valid date format
 */
export function getReadableDate(inputDate) {
  const nth = (d) => {
    if (d > 3 && d < 21) return 'th';

    switch (d % 10) {
      case 1:
        return 'st';
      case 2:
        return 'nd';
      case 3:
        return 'rd';
      default:
        return 'th';
    }
  };

  const myDate = new Date(inputDate);
  const date = myDate.getDate();
  const month = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ][myDate.getMonth()];

  return `${date}${nth(date)} ${month} ${myDate.getFullYear()}`;
}

// string -> string
export function prepareDate(date = '') {
  if (!date) return;
  const [day, month, year] = date.toString().split('/');

  // eslint-disable-next-line consistent-return
  return [month, day, year].join('/');
}
