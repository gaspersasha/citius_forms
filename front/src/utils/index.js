export * from './cms';
export * from './currency';
export * from './date';
export * from './form';
export * from './gtm';
export * from './network';
export * from './objects';
export * from './script';
export * from './validation';
export * from './text';
export * from './scroll-up-to';
export * from './permutive';
export * from './vehicle-data';
export * from './amp-styles';
export * from './prevent-body-scroll';
export * from './polyfill';
export * from './go';
export * from './state';
export * from './get-cookie';
