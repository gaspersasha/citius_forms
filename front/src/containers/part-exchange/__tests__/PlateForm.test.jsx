import React from 'react';

import { endpointFetch } from '../../../utils';
import PlateForm from '../PlateForm';

const props = {
  payData: {},
  setData: jest.fn(),
  setNextStep: jest.fn(),
  urlParams: { productAdvertId: '' },
  productAdvertId: '',
  deleteEmptyUrlParams: jest.fn(),
  scroll: jest.fn(),
  leadgen: '0',
  setIsLoading: jest.fn(),
};

const submitEvent = {
  preventDefault: jest.fn(),
};

jest.mock('~utils', () => ({
  ...jest.requireActual('../../../utils'),
  endpointFetch: jest.fn(),
  endpointPush: jest.fn(),
}));

endpointFetch.mockResolvedValue();

describe('<PlateForm />', () => {
  let wrapper;
  const getWrapper = (method = shallow) => method(<PlateForm {...props} />);

  it('PlateForm snapshot', async () => {
    wrapper = getWrapper();
    await global.updateWrapper(wrapper);
    expect(wrapper).toMatchSnapshot();
  });

  it('plate input responds to changes', () => {
    wrapper = getWrapper(mount);
    const plate = 'input[type="text"][name="plate"]';
    const event = { target: { value: 'AB12CDE', name: 'plate' } };

    wrapper.find(plate).simulate('change', event);
    expect(wrapper.find(plate).props().value).toEqual('AB12CDE');
  });

  it('invalid plate input not responds to changes', () => {
    wrapper = getWrapper(mount);
    const plate = 'input[type="text"][name="plate"]';
    const event = { target: { value: 'AB12CDE213323', name: 'plate' } };

    wrapper.find(plate).simulate('change', event);
    expect(wrapper.find(plate).props().value).toEqual(undefined);
  });

  it('valid form can be submitted', async () => {
    wrapper = getWrapper(mount);
    await global.updateWrapper(wrapper);
    const plate = 'input[type="text"][name="plate"]';
    const event = { target: { value: 'AB12CDE', name: 'plate' } };

    wrapper.find(plate).simulate('change', event);
    wrapper.find('form').simulate('submit', submitEvent);
    expect(endpointFetch).toHaveBeenCalledWith('checkPlatePartEx', {
      ...props.deleteEmptyUrlParams(props.urlParams),
      ...props.payData,
    });
  });
});
