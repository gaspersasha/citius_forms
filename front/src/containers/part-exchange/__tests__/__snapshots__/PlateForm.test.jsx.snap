// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlateForm /> PlateForm snapshot 1`] = `
<form
  data-id="plate-form"
  onSubmit={[Function]}
>
  <h5
    className="heading"
  >
    Add part exchange?
  </h5>
  <h5
    className="description"
  >
    Enter your reg below to get your estimated part exchange price
  </h5>
  <div
    className="plate"
  >
    <InputText
      className=""
      customClasses=""
      errorMessage=""
      label=""
      name="plate"
      onChange={[Function]}
      placeholder="e.g. AB12CDE"
      required={false}
      status="default"
      subTitle=""
    />
  </div>
  <p
    className="infoBox"
  >
    By using this service, you agree that your vehicle registration details will be passed to a third party, in order to look up the vehicle details.
  </p>
  <div
    className="alignCenter"
  >
    <Button
      className=""
      disabled={false}
      href=""
      name="partex__submit-btn"
      onClick={null}
      selected={false}
      styleType="primary"
      type="submit"
    >
      Next
    </Button>
    <span
      className="skipPX"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      Skip part exchange
    </span>
  </div>
</form>
`;
